extension/llm/runner/irunner.h

```diff
@@ -6,42 +6,114 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-// An interface for LLM runners. Developers can create their own runner that
-// implements their own load and generation logic to run the model.
+// Interface for text generation runners.
 
 #pragma once
 
+#include <cstdint>
 #include <functional>
+#include <memory>
 #include <string>
 
 #include <executorch/extension/llm/runner/stats.h>
-#include <executorch/extension/module/module.h>
+#include <executorch/runtime/core/error.h>
 
 namespace executorch {
 namespace extension {
 namespace llm {
 
+// Configuration struct for generation parameters
+struct GenerationConfig {
+  // Temperature for sampling (higher = more random)
+  float temperature = 0.8f;
+
+  // Whether to echo the input prompt in the output
+  bool echo = true;
+
+  // Maximum number of total tokens
```

Comment:
More doc string on this: a limit that users can set. It must not be larger than max_context_len from the model. (something like that, if I understand it correctly).